### PR TITLE
Updated interpolator for vii tie points for test data version V2 

### PR DESCRIPTION
**Pull request by pepephillips**
- [ ] Tests added

### CODE REVIEW

File: geotiepoints/viiinterpolator.py [hunk 1]
```diff
@@ -64,39 +65,57 @@ def tie_points_interpolation(data_on_tie_points, scan_alt_tie_points, tie_points
...
+                         coords={'num_tie_points_alt': pix_alt, 'num_tie_points_act': pix_act})
+        data_on_pixel_points_granule.attrs = combine_metadata(data)
+
+        # loop over scans
+        for j_scan in range(n_scans):
+            index_tie_points_start = j_scan * scan_alt_tie_points
+            index_tie_points_end = (j_scan * scan_alt_tie_points) + scan_alt_tie_points
+            index_pixel_start = j_scan * (scan_alt_tie_points - 1) * tie_points_factor
+            index_pixel_end = (j_scan * (scan_alt_tie_points - 1) * tie_points_factor) + \
+                              (scan_alt_tie_points - 1) * tie_points_factor
```
Comment (mraspaud) [inline geotiepoints/viiinterpolator.py:100]: Do I understand correctly that this has been moved inside the granule loop and is now being repeate for every array?
What is the performance hit of this?
Also have you tried vectorizing this loop?

**Review by mraspaud — COMMENTED** [review]
Thanks for the update on vii interpolation for the test data v2!
I'm a bit worried about the inner loop (for scans), see inline comment.

File: geotiepoints/viiinterpolator.py (hunk 1 above)
Comment (pepephillips) [reply geotiepoints/viiinterpolator.py]: There is over a factor of 10 hit on performance so I have reverted to the original methodology which interpolates the granule as a whole, although not strictly accurate. We will investigate the impact of this on geolocation accuracy further but for now I think its the most practical solution.

**Comment by wolterse** [comment]
Hi there, can you give me any projection on when the updated vii geotie interpolation for the V2 synthetic data will be released?
Thanks for any information!

**Comment by mraspaud** [comment]
@pepephillips ^

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**